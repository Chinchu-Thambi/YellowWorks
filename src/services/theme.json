{
  "colors": {
    "base": {
      "100": "#FFFFFF",
      "200": "#F4F4F4",
      "300": "#F2F2F2",
      "400": "#E7E7E7"
    },
    "brand": {
      "100": "#FFFDF4",
      "200": "#FEF9D3",
      "300": "#FDF3A7",
      "400": "#FDED7A",
      "500": "#FCE74E",
      "600": "#FBE122"
    },
    "contrast": {
      "100": "#F3F5F8",
      "200": "#D2D4DA",
      "300": "#A5AAB4",
      "400": "#797F8F",
      "500": "#4C5569",
      "600": "#1F2A44"
    },
    "crimson": {
      "100": "#FDF5F5",
      "200": "#F7D9D8",
      "300": "#EFB2B1",
      "400": "#E88C8B",
      "500": "#E06564",
      "600": "#D83F3D"
    },
    "fern": {
      "100": "#F7FBF6",
      "200": "#E0EFDD",
      "300": "#C1DFBB",
      "400": "#A2D09A",
      "500": "#83C078",
      "600": "#64B056"
    },
    "uplift": {
      "100": "#FEF8F6",
      "200": "#FBE2DA",
      "300": "#F8C6B5",
      "400": "#F4A991",
      "500": "#F18D6C",
      "600": "#ED7047"
    },
    "pacific": {
      "100": "#F7FBFE",
      "200": "#E1F2FA",
      "300": "#C4E5F5",
      "400": "#A6D8F1",
      "500": "#89CBEC",
      "600": "#6BBEE7"
    },
    "magnificent": {
      "100": "#F7F5FA",
      "200": "#E2D8EA",
      "300": "#C5B0D5",
      "400": "#A789C0",
      "500": "#8A61AB",
      "600": "#6D3A96"
    },
    "violet": {
      "100": "#F7F7FA",
      "200": "#DFE1EC",
      "300": "#C0C2D8",
      "400": "#A0A4C5",
      "500": "#8185B1",
      "600": "#61679E"
    }
  },
  "opacity": {
    "95": "0.95"
  },
  "spacing": {
    "px": "1px",
    "0": "0rem",
    "1": "0.3125rem",
    "2": "0.625rem",
    "2.5": "1rem",
    "3": "1.25rem",
    "4": "2rem",
    "5": "3rem",
    "6": "4rem",
    "7": "5rem",
    "8": "6rem",
    "9": "7rem",
    "10": "8rem",
    "11": "10rem",
    "12": "12rem",
    "13": "14rem",
    "14": "16rem",
    "15": "18rem",
    "16": "20rem",
    "17": "24rem",
    "18": "32rem",
    "19": "36rem",
    "20": "48rem"
  },
  "screens": {
    "sm": "36em",
    "md": "48em",
    "lg": "62em",
    "xl": "75em",
    "2xl": "90em"
  },
  "borderRadius": {
    "none": "0",
    "sm": "0.125rem",
    "default": "0.15625rem",
    "md": "0.3125rem",
    "lg": "0.625rem",
    "xl": "0.75rem",
    "2xl": "1rem",
    "3xl": "1.5rem",
    "full": "9999px"
  },
  "boxShadow": {
    "xs": "0 0 0 1px rgb(31, 42, 68, 0.05)",
    "sm": "0 1px 2px 0 rgb(31, 42, 68, 0.05)",
    "default": "0 1px 3px 0 rgb(31, 42, 68, 0.1), 0 1px 2px 0 rgb(31, 42, 68, 0.06)",
    "md": "0 4px 6px -1px rgb(31, 42, 68, 0.1), 0 2px 4px -1px rgb(31, 42, 68, 0.06)",
    "lg": "0 10px 15px -3px rgb(31, 42, 68, 0.1), 0 4px 6px -2px rgb(31, 42, 68, 0.05)",
    "xl": "0 20px 25px -5px rgb(31, 42, 68, 0.1), 0 10px 10px -5px rgb(31, 42, 68, 0.04)",
    "2xl": "0 25px 50px -12px rgb(31, 42, 68, 0.25)",
    "inner": "inset 0 2px 4px 0 rgba(0, 0, 0, 0.06)",
    "outline": "0 0 0 3px rgba(66, 153, 225, 0.5)",
    "none": "none"
  },
  "fontFamily": {
    "title": [
      "Montserrat",
      "Lato",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "Segoe UI",
      "Roboto",
      "Helvetica Neue",
      "Arial",
      "Noto Sans",
      "sans-serif"
    ],
    "sans": [
      "Lato",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "Segoe UI",
      "Roboto",
      "Helvetica Neue",
      "Arial",
      "Noto Sans",
      "sans-serif"
    ],
    "serif": [
      "Times",
      "serif"
    ],
    "mono": [
      "monospace"
    ]
  },
  "fontWeight": {
    "normal": "400",
    "bold": "700"
  },
  "fontSize": {
    "xs": [
      "0.75rem",
      {
        "lineHeight": "1rem"
      }
    ],
    "sm": [
      "0.875rem",
      {
        "lineHeight": "1.25rem"
      }
    ],
    "base": [
      "1rem",
      {
        "lineHeight": "1.5rem"
      }
    ],
    "lg": [
      "1.125rem",
      {
        "lineHeight": "1.75rem"
      }
    ],
    "xl": [
      "1.25rem",
      {
        "lineHeight": "1.75rem"
      }
    ],
    "2xl": [
      "1.5rem",
      {
        "lineHeight": "2rem"
      }
    ],
    "3xl": [
      "1.875rem",
      {
        "lineHeight": "2.25rem"
      }
    ],
    "4xl": [
      "2.25rem",
      {
        "lineHeight": "2.5rem"
      }
    ],
    "5xl": [
      "3rem",
      {
        "lineHeight": "1"
      }
    ],
    "6xl": [
      "3.75rem",
      {
        "lineHeight": "1"
      }
    ],
    "7xl": [
      "4.5rem",
      {
        "lineHeight": "1"
      }
    ],
    "8xl": [
      "6rem",
      {
        "lineHeight": "1"
      }
    ],
    "9xl": [
      "8rem",
      {
        "lineHeight": "1"
      }
    ]
  }
}