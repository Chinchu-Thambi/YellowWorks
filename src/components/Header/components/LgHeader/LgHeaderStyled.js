import React from 'react';
import PropTypes from 'prop-types';

import { Box } from 'rebass/styled-components';
import { Link } from 'gatsby';
import styled from 'styled-components';
import { mediaBreakpointUp } from '../../../../util';
import Button from '../../../Button';
import theme from '../../../../util/theme';

const propTypeModels = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  theme: PropTypes.shape({
    containerWidth: PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    ),
  }),
};

export const SkipLink = styled.div`
  position: relative;
  font-family: ${theme.fonts.mono};
  font-size: ${theme.fontSizes[1]};
  text-transform: uppercase;

  button {
    height: 1px;
    padding: ${theme.space[1]} ${theme.space[2]};
    position: absolute;
    left: -100%;
    overflow: hidden;
    top: -100%;
    width: 1px;
    text-decoration: none;
  }

  button:active, button:focus {
    display: inline-block;
    padding: ${theme.space[3]};
    height: auto;
    left: 0;
    margin: ${theme.space[3]};
    overflow: visible;
    position: absolute;
    top: 85px;
    width: auto;
    outline: none;
    z-index: 10;
  }
`;

export const Container = styled(Box)`
  display: none;
  position: relative;
  z-index: ${theme.zIndex.sticky};
  height: 100%;
  align-items: center;
  background-color: ${theme.palette.base[0]};
  box-shadow: ${theme.shadows.small};

  ${mediaBreakpointUp('md')} {
    display: block;
  }

  li {
    list-style: none;
  }

  a {
    text-decoration: none;
  }
`;

export const WhiteNav = ({ className, children }) => (
  <div className={`local-reset h-7 shadow ${className}`}>
    <ul className="container px-2 mx-auto flex justify-around items-center h-full bg-base-100">
      {children}
    </ul>
  </div>
);

WhiteNav.defaultProps = {
  className: '',
};

WhiteNav.propTypes = {
  className: PropTypes.string,
  children: propTypeModels.children.isRequired,
};

export const HeaderButton = styled(Button)`
  height: auto;
  font-size: ${theme.fontSizes[1]};
  font-weight: ${theme.fontWeight[1]};
  color: ${theme.palette.contrast[1]};
  padding: ${theme.space[2]} ${theme.space[3]};
`;

export const MenuLink = styled(Link)`
  position: relative;

  ::after {
    content: '';
    display: block;
    height: 6px;
    border-radius: ${theme.space[3]};
    background: ${theme.palette.brand[0]};
    position: absolute;
    bottom: 16px;
    width: 0;
    left: 50%;
    transition: width .2s, left .2s;
  }

  :hover::after {
    width: calc(100% - 2 * ${theme.space[3]});
    left: ${theme.space[3]};
  }
`;

export const ChildMenuWrapper = styled.div`
  background: ${theme.colors.white};
  position: absolute;
  transition: transform .2s;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: -10;

  transform: ${({ visible }) => (
    visible
      ? 'translateY(100%)'
      : 'translateY(-90px)'
  )};
}
`;
