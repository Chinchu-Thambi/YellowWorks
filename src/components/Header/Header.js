import React from 'react';
import Loadable from 'react-loadable';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';

import SmHeader from './components/SmHeader';
import LgHeader from './components/LgHeader';
import AuthModal from '../Auth/Modal/AuthModal';
import AuthContext from '../Auth/AuthContext';

const CustomerStatusBarSwitch = Loadable({
  loader: () => import('../CustomerStatusBarSwitch'),
  loading: () => null,
});

const Header = () => {
  const { isStaff } = React.useContext(AuthContext) || {};

  return (
    <>
      <SmHeader />
      <LgHeader />
      <AuthModal />
      <NotificationContainer />

      {isStaff && (
        <CustomerStatusBarSwitch />
      )}
    </>
  );
};

export default Header;
