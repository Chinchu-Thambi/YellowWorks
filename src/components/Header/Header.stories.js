import React from 'react';
import { ThemeProvider } from 'styled-components';
import SmHeader from './SmHeader';
import LgHeader from './LgHeader';
import { AuthProvider } from '../Auth/AuthContext';
import { theme } from '../../util';

export const Header = () => (
  <AuthProvider>
    <ThemeProvider theme={theme}>
      <SmHeader />
      <LgHeader />
    </ThemeProvider>
  </AuthProvider>
);

export default {
  title: 'Components',
};
