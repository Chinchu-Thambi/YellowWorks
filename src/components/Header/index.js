import React from 'react';
import Loadable from 'react-loadable';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';

import SmHeader from './SmHeader';
import LgHeader from './LgHeader';
import AuthModal from '../Auth/Modal/AuthModal';
import AuthContext from '../Auth/AuthContext';

const CustomerStatusBarSwitch = Loadable({
  loader: () => import('../CustomerStatusBarSwitch'),
  loading: () => null,
});

const Header = () => {
  const { isStaff } = React.useContext(AuthContext) || {};

  return (
    <>
      <SmHeader />
      <LgHeader />
      <AuthModal />
      <NotificationContainer />

      {isStaff && (
        <CustomerStatusBarSwitch />
      )}
    </>
  );
};

export default Header;
