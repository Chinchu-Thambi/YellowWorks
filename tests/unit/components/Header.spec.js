/* globals jest, describe, it, expect */

import { mount, shallow } from 'enzyme';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../../../src/util/theme';

import AuthContext from '../../../src/components/Auth/AuthContext';

import Header from '../../../src/components/Header';
import { MenuLink, ProductMenu } from '../../../src/components/Header/LgHeaderStyled';

const authValue = {
  state: {
    user: {
      attributes: {
        given_name: '',
        family_name: '',
      },
      username: 'user',
    },
    status: 'AUTHENTICATED',
    modal: {},
  },
  jwtToken: 'some_token',
  initModal: () => {},
  LoginLink: () => null,
};


describe('Header', () => {
  it('exports a header matching the previous snapshot', () => {
    const wrapper = mount(
      <AuthContext.Provider value={authValue}>
        <ThemeProvider theme={theme}>
          <Header />
        </ThemeProvider>
      </AuthContext.Provider>,
    );
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.unmount();
  });

  it('toggles product menu visibility', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    const wrapper = mount(
      <AuthContext.Provider value={authValue}>
        <ThemeProvider theme={theme}>
          <Header />
        </ThemeProvider>
      </AuthContext.Provider>,
    );

    const ourProductsLink = wrapper.find(MenuLink).find({
      to: '/our-products/',
    }).first();

    ourProductsLink.simulate('mouseenter');
    expect(setState).toHaveBeenCalledWith(true);
    setState.mockClear();

    // mouse leaves to productMenu, doesn't toggle it to false
    ourProductsLink.simulate('mouseleave', {
      relatedTarget: {
        className: ProductMenu.styledComponentId,
      },
    });
    expect(setState).not.toHaveBeenCalled();

    // mouse leaves to anything else, toggles productMenu to true
    ourProductsLink.simulate('mouseleave', {
      relatedTarget: {
        className: 'anything else',
      },
    });
    expect(setState).toHaveBeenCalledWith(false);
    setState.mockClear();

    const productMenu = wrapper.find(ProductMenu).first();
    productMenu.simulate('mouseleave');
    expect(setState).toHaveBeenCalledWith(false);

    wrapper.unmount();
    useStateSpy.mockRestore();
  });

  it('transforms ProductMenu when toggling visibility', () => {
    const hidden = shallow(<ProductMenu theme={theme} />).html();
    const visible = shallow(<ProductMenu visible theme={theme} />).html();
    expect(hidden).not.toEqual(visible);
  });
});
