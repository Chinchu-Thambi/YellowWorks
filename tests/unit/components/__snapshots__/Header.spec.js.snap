// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header exports a header matching the previous snapshot 1`] = `"<div class=\\"sc-hBMVcZ iZbdK\\"><div class=\\"sc-fnVYJo iVQQrb\\"><div class=\\"sc-fFSRdu jXiHeG\\"><img src=\\"test-file-stub\\" alt=\\"Digital Marketing Solutions by Yellow\\"></div><a href=\\"/\\"><img alt=\\"Yellow NZ\\" src=\\"test-file-stub\\"> </a></div><div class=\\"sc-bkbjAj hNHvBj\\"><ul><div class=\\"sc-iCoHVE cxpYLP\\"><section class=\\"sc-fujyUd dSjaGg\\"><header><h4>My account</h4></header><ul><li><a class=\\"sc-pNWxx sc-jrsJCI ixhjXK czgIIF\\" href=\\"/my-yellow/account-settings\\">Account settings</a></li><li><a class=\\"sc-pNWxx sc-jrsJCI ixhjXK czgIIF\\" href=\\"/my-yellow/my-products\\">My products</a></li><li><a class=\\"sc-pNWxx sc-jrsJCI ixhjXK czgIIF\\" href=\\"/my-yellow/insights\\">Insights</a></li></ul></section></div><li class=\\"sc-iemXMA cGUNux\\"><a href=\\"/\\">Home</a></li><li class=\\"sc-dlMBXb lNdUQ\\"><div><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products\\" href=\\"/our-products\\">Products</a><button aria-label=\\"Expand menu\\" class=\\"sc-dlnjPT liGRPc\\"></button></div><ul><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Yellow_Profile\\" href=\\"/our-products/yellow-profile/\\">Yellow Profile</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_SEO\\" href=\\"/our-products/seo/\\">Search Engine Optimization</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Adwords\\" href=\\"/our-products/google-search-ads/\\">Google Search Ads</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Display_Advertising\\" href=\\"/our-products/google-display-ads/\\">Google Display Ads</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Facebook_Adwords\\" href=\\"/our-products/facebook-ads/\\">Facebook Ads</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Marketing_Automation\\" href=\\"/our-products/marketing-automation/\\">Marketing Automation &amp; CRM</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Yellow_Pages\\" href=\\"/our-products/yellow-pages/\\">Yellow Pages</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_White_Pages\\" href=\\"/our-products/white-pages/\\">White Pages</a></li></ul></li><li class=\\"sc-dlMBXb lNdUQ\\"><div><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources\\" href=\\"/resources/\\">Resources</a><button aria-label=\\"Expand menu\\" class=\\"sc-dlnjPT liGRPc\\"></button></div><ul><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Customer_Stories\\" href=\\"/resources/customer-stories/\\">Customer Stories</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Tips_and_Tools\\" href=\\"/resources/tips-and-tools/\\">Tips and Tools</a></li><li><a data-ga=\\"track\\" data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_ebook\\" href=\\"/resources/yellow-ebook/\\">Yellow eBook</a></li><li><a href=\\"https://hub.yellow.co.nz/knowledge/faq\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"FAQ\\">FAQ</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Flybuys\\" href=\\"/resources/flybuys/\\">Flybuys</a></li></ul></li><li class=\\"sc-dlMBXb lNdUQ\\"><div><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"About Yellow\\" href=\\"/company/about-us/\\">About Yellow</a><button aria-label=\\"Expand menu\\" class=\\"sc-dlnjPT liGRPc\\"></button></div><ul><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Our_Team\\" href=\\"/company/about-us/\\">Our team</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Careers\\" href=\\"/company/careers/\\">Careers</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Contact_Us\\" href=\\"/company/contact-us/\\">Contact us</a></li></ul></li><li class=\\"sc-iemXMA cGUNux\\"><a href=\\"/find-a-business/\\"><span class=\\"sc-dIvqjp emwODp\\">Find a Business</span></a></li></ul><ul class=\\"sc-kfYqjs gBXWRw\\"><li><button class=\\"sc-dlnjPT liGRPc sc-iwaifL vHyDA\\"><svg viewBox=\\"0 0 30 30\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M25.6059 4.39361C22.7729 1.56001 19.0062 0 14.9994 0C10.9926 0 7.22642 1.56001 4.39352 4.39361C1.55998 7.22658 0 10.9934 0 14.9997C0 19.006 1.55998 22.7728 4.39352 25.6064C7.22642 28.4394 10.9932 30 15 30C19.0068 30 22.773 28.44 25.6065 25.6064C28.4394 22.7734 30 19.0066 30 14.9997C30 10.9928 28.4388 7.22658 25.6059 4.39361ZM22.9328 25.4416C20.7279 27.1207 17.9783 28.1191 14.9994 28.1191C12.0204 28.1191 9.27025 27.1207 7.06598 25.4409C6.70497 25.1658 6.54891 24.6926 6.69369 24.262C7.46522 21.9605 9.3135 20.0971 11.687 19.184C12.0066 19.0611 12.0699 18.6324 11.7929 18.4312C10.4228 17.4365 9.52973 15.822 9.52973 14.0025C9.52973 10.9144 12.1019 8.41617 15.2162 8.53651C18.1375 8.64995 20.469 11.078 20.4696 14.0013C20.4703 15.8207 19.5772 17.4353 18.2065 18.4306C17.9294 18.6318 17.9927 19.0605 18.3124 19.1833C20.6777 20.0934 22.521 21.9467 23.2975 24.2369C23.4461 24.6769 23.3013 25.1608 22.9328 25.4416Z\\" fill=\\"#A5AAB4\\"></path></svg>Sign Out</button></li><li><div><a href=\\"tel:0800803803\\"><img alt=\\"Expand menu\\" src=\\"test-file-stub\\">Call us</a></div></li></ul><div class=\\"sc-bCwgka okBeA\\"><h3>Follow us</h3><ul><li><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"//www.facebook.com/yellownz\\" title=\\"Follow us on Facebook\\" data-ga=\\"track\\" data-ga-cat=\\"Follow_Us\\" data-ga-act=\\"Facebook\\" data-ga-lab=\\"Facebook\\"><img src=\\"test-file-stub\\" alt=\\"Facebook\\"></a></li><li><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"//twitter.com/yellownz\\" title=\\"Follow us on Twitter\\" data-ga=\\"track\\" data-ga-cat=\\"Follow_Us\\" data-ga-act=\\"Twitter\\" data-ga-lab=\\"Twitter\\"><img src=\\"test-file-stub\\" alt=\\"Twitter\\"></a></li><li><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"//www.linkedin.com/company/yellow-new-zealand\\" title=\\"Follow us on LinkedIn\\" data-ga=\\"track\\" data-ga-cat=\\"Follow_Us\\" data-ga-act=\\"LinkedIn\\" data-ga-lab=\\"LinkedIn\\"><img src=\\"test-file-stub\\" alt=\\"LinkedIn\\"></a></li><li><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"//www.youtube.com/user/yellowbiznz\\" title=\\"Follow us on Youtube\\" data-ga=\\"track\\" data-ga-cat=\\"Follow_Us\\" data-ga-act=\\"Youtube\\" data-ga-lab=\\"Youtube\\"><img src=\\"test-file-stub\\" alt=\\"YouTube\\"></a></li><li><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://www.google.co.nz/maps/place/Yellow/@-36.897607,174.8006043,17z/data=!3m1!4b1!4m5!3m4!1s0x6d0d48ef0f099ad1:0xbabdfef7f73a92a5!8m2!3d-36.897607!4d174.802793\\" title=\\"Google Maps\\" data-ga=\\"track\\" data-ga-cat=\\"Follow_Us\\" data-ga-act=\\"Maps\\" data-ga-lab=\\"Maps\\"><img src=\\"test-file-stub\\" alt=\\"Google\\"></a></li></ul></div></div></div><div class=\\"sc-bdnylx sc-lmgQde kfHHiO hbMDAh\\"><div class=\\"sc-cxNIbT cWFJbw\\"><button class=\\"sc-dlnjPT liGRPc\\">Skip to main content</button></div><div width=\\"1,33.75rem,43.75rem,58.75rem,73.75rem,90.00rem\\" class=\\"sc-bdnylx sc-gtssRu sc-giAqnE dnYBaK kyjsSx dPBXdQ\\"><a href=\\"/\\"><img src=\\"https://cdn11.bigcommerce.com/s-npte00i8ef/stencil/4a56d7a0-2da2-0137-892b-0242ac110005/e/2ebf55a0-80d9-0136-d2c6-7fd6dfdde2be/img/header-footer/yellow-logo.svg\\" alt=\\"Yellow\\"></a><ul class=\\"sc-bdnylx sc-gtssRu sc-iJCRLp dhteoD kyjsSx gkoQmp\\"><li><a class=\\"sc-kLokih fpcqPB\\" href=\\"/our-products/\\">Products<img class=\\"pointer\\" src=\\"test-file-stub\\" alt=\\"\\" css=\\"[object Object]\\"></a><div class=\\"sc-iklIKw japDYA\\"><div width=\\"1,33.75rem,43.75rem,58.75rem,73.75rem,90.00rem\\" class=\\"sc-bdnylx sc-gtssRu llWFtj kyjsSx ProductMenuFlex\\"><ul><li><p>Get Online</p></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Yellow_Profile\\" href=\\"/our-products/yellow-profile/\\">Yellow Profile</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_SEO\\" href=\\"/our-products/seo/\\">Search Engine Optimisation</a></li></ul><ul><li><p>Get More Customers</p></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Adwords\\" href=\\"/our-products/google-search-ads/\\">Google Search Ads</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Display_Advertising\\" href=\\"/our-products/google-display-ads/\\">Google Display Ads</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Facebook_Adwords\\" href=\\"/our-products/facebook-ads/\\">Facebook Ads</a></li></ul><ul><li><p>Do Business Better</p></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Marketing_Automation\\" href=\\"/our-products/marketing-automation/\\">Marketing Automation &amp; CRM</a></li></ul><ul><li><p>Print Directories</p></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_Yellow_Pages\\" href=\\"/our-products/yellow-pages/\\">Yellow Pages</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Products_White_Pages\\" href=\\"/our-products/white-pages/\\">White Pages</a></li></ul></div></div></li><li><a class=\\"sc-kLokih fpcqPB\\" href=\\"/resources/\\">Resources<img class=\\"pointer\\" src=\\"test-file-stub\\" alt=\\"\\" css=\\"[object Object]\\"></a><div class=\\"sc-iklIKw jwbMIR\\"><div width=\\"1,33.75rem,43.75rem,58.75rem,73.75rem,90.00rem\\" class=\\"sc-bdnylx sc-gtssRu llWFtj kyjsSx ProductMenuFlex\\"><ul><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Customer_Stories\\" href=\\"/resources/customer-stories/\\">Customer Stories</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Tips_and_Tools\\" href=\\"/resources/tips-and-tools/\\">Tips and Tools</a></li><li><a data-ga=\\"track\\" data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_ebook\\" href=\\"/resources/yellow-ebook/\\">Yellow eBook</a></li><li><a href=\\"https://hub.yellow.co.nz/knowledge/faq\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Knowledge_Base\\">FAQ</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"Resources_Flybuys\\" href=\\"/resources/flybuys/\\">Flybuys</a></li></ul></div></div></li><li><a class=\\"sc-kLokih fpcqPB\\" href=\\"/company/about-us/\\">About Yellow<img class=\\"pointer\\" src=\\"test-file-stub\\" alt=\\"\\" css=\\"[object Object]\\"></a><div class=\\"sc-iklIKw cVyRWb\\"><div width=\\"1,33.75rem,43.75rem,58.75rem,73.75rem,90.00rem\\" class=\\"sc-bdnylx sc-gtssRu llWFtj kyjsSx ProductMenuFlex\\"><ul><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"About_Our_team\\" href=\\"/company/about-us/\\">Our Team</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"About_Careers\\" href=\\"/company/careers/\\">Careers</a></li><li><a data-ga-cat=\\"Navigation_Interaction\\" data-ga-act=\\"Header_Clicks\\" data-ga-lab=\\"About_Contact_Us\\" href=\\"/company/contact-us/\\">Contact Us</a></li></ul></div></div></li><li><a href=\\"/find-a-business/\\" class=\\"sc-dlnjPT liGRPc sc-ezzayL fJuNId\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"search\\" class=\\"svg-inline--fa fa-search fa-w-16 fa-sm \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z\\"></path></svg><div class=\\"css-1k3micg\\">Find a business</div></a></li><li class=\\"sc-bYwzba gzLpPq\\"></li><li><button id=\\"account-button\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" data-reach-menu-button=\\"true\\" type=\\"button\\" class=\\"sc-hiKfjK juoqFF\\"><img src=\\"test-file-stub\\" alt=\\"account\\"><div class=\\"user-name\\"> </div></button></li></ul></div><div tabindex=\\"-1\\"></div></div><div class=\\"notification-container notification-container-empty\\"><div></div></div>"`;
